packages/components/src/color-palette/test/index.tsx

```diff
@@ -201,4 +212,41 @@ describe( 'ColorPalette', () => {
 			screen.getByRole( 'button', { name: 'Clear' } )
 		).toBeInTheDocument();
 	} );
+
+	it( 'should display the selected color name and value', async () => {
+		const user = userEvent.setup();
+
+		const { container } = render( <UnmockedColorPalette /> );
+
+		expect( screen.getByText( 'No color selected' ) ).toBeVisible();
+		expect(
+			// Relying on implementation details here, but it's the only way I could come up with.
+			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+			container.querySelector(
+				'.components-color-palette__custom-color-value'
+			)
+		).toHaveTextContent( '' );
+
+		// Click the first unpressed button
+		await user.click(
+			screen.getAllByRole( 'button', {
+				name: /^Color:/,
+				pressed: false,
+			} )[ 0 ]
+		);
+
+		expect( screen.getByText( EXAMPLE_COLORS[ 0 ].name ) ).toBeVisible();
+		expect( screen.getByText( EXAMPLE_COLORS[ 0 ].color ) ).toBeVisible();
+
+		// Clear the color, confirm that name and value are correctly updated
+		await user.click( screen.getByRole( 'button', { name: 'Clear' } ) );
+		expect( screen.getByText( 'No color selected' ) ).toBeVisible();
+		expect(
+			// Relying on implementation details here, but it's the only way I could come up with.
+			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+			container.querySelector(
+				'.components-color-palette__custom-color-value'
+			)
+		).toHaveTextContent( '' );
```

Comment:
Here you could check that `screen.queryByText( EXAMPLE_COLORS[ 0 ].name )` and that `screen.queryByText( EXAMPLE_COLORS[ 0 ].color )` are not in the document anymore?

Reply:
```suggestion
```

Given the new checks, we can probably remove this one?